File: Cuadrado.java


  

  public class Cuadrado extends Rectangulo {
    
    public Cuadrado(int lado) {
      super(lado,1);
      this.ancho=this.alto;
    }
    
   
    public void setLado(int lado) {
      super.setAlto(lado);
      super.setAncho(lado);

    }
    
    
      
    
    /**
     * Devuelve un boolean verdadero si los cuadrados son iguales y falso en caso contrario
     * @param cuad
     * @return
     */
    public boolean equals(Cuadrado cuad) {
      if(this.alto==cuad.alto)
        return true;
      else
        return false;
      
    }
    /**
     * Compara dos cuadrados y devuelve 0 si son iguales, -1 si el argumento es mayor y 1 si el
     * principal es mayor
     * @param cuad
     * @return
     */
    public int compareTo(Cuadrado cuad) {
      if(this.alto==cuad.alto)
        return 0;
      else if (this.alto>cuad.alto) {
        return 1;
        }
      else
        return -1;
      
    }
    
  }

File: Cuadrado.py
'''
Created on 5 abr. 2019

@author: d18sisaj
'''
from Rectangulo import Rectangulo

class Cuadrado(Rectangulo):
    
    
    def __init__(self,lado):
        Rectangulo.__init__(self,lado,lado)
        
    @property   
    def lado(self):
        return self._alto
   
    '''NO RECONOCE LA PROPIEDAD EL SETER'''
    @lado.setter   
    def lado(self, lado):
        Rectangulo.alto(lado)
        Rectangulo.ancho(lado)
    
    def __eq__(self, cuadrado):
        if(self._alto==cuadrado._alto):
            return True
        else:
            return False

    def __gt__(self,cuadrado):
        if(self._alto>cuadrado._alto):
            return True
        else:
            return False
    
    def __lt__(self,cuadrado):
        if(self._alto<cuadrado._alto):
            return True
        else:
            return False
    
    def __ge__(self,cuadrado):
        if(self._alto>=cuadrado._alto):
            return True
        else:
            return False
        
    def __le__(self,cuadrado):
        if(self._alto<=cuadrado._alto):
            return True
        else:
            return False    